### PR TITLE
Feature: implement Tabs component

### PR DESCRIPTION
**Pull request by akhlopiachyi**
* Feature: add Tab component
* Feature: add Tabs component

### CODE REVIEW

File: src/components/tab/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    children,
+    className,
+    onClick,
+    tabIndex,
+    active,
+    disabled,
+    linkClassName,
+}) => {
+    const cnLink = classNames(`tab-link ${linkClassName}`, {
+        'active': active,
```
Comment (berezh) [inline src/components/tab/index.tsx]: use BEM in all styles. Check out example in https://github.com/openware/react-components/tree/master/src/components/landing-block

File: src/components/index.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
 export * from './button';
 export * from './landing-block';
 export * from './icon';
+export * from './tab';
```
Comment (berezh) [inline src/components/index.ts]: for one component one export, even you exporting many components regarding your component

**Review by berezh — CHANGES_REQUESTED** [review]
Build is failed

File: src/components/tabs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import classNames from 'classnames';
+import React, { useState, useEffect } from 'react';
+import { Tab } from './tab';
+
+import './index.scss';
+
+interface TabsProps {
+    children: any;
```
Comment (berezh) [inline src/components/tabs/index.tsx]: type should be `React.ReactNode` or other, but not `any`

File: src/components/tabs/index.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+}) => {
+    const [activeTabIndex, setActiveTab] = useState(0);
+
+    useEffect(() => {
+        if (selectedTabIndex && selectedTabIndex > -1 && children[selectedTabIndex]) {
+            setActiveTab(selectedTabIndex);
+        }
+    });
+
+    const handleClick = (index: number) => {
```
Comment (berezh) [inline src/components/tabs/index.tsx]: `useCallback` is missing

File: src/components/tabs/index.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    };
+
+    const renderActiveTabContent = () => {
+        if (children[activeTabIndex]) {
+            return children[activeTabIndex].props.panel;
+        }
+    }
+
+    return (
+        <>
```
Comment (berezh) [inline src/components/tabs/index.tsx]: extra tag

File: src/components/tabs/index.tsx (hunk 3 above)
Comment (akhlopiachyi) [reply src/components/tabs/index.tsx]: Done

File: src/components/tabs/index.tsx (hunk 4 above)
Comment (akhlopiachyi) [reply src/components/tabs/index.tsx]: Done

File: src/components/tabs/index.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        if (
+            selectedTabIndex &&
+            selectedTabIndex > -1 &&
+            children[selectedTabIndex]
+        ) {
+            setActiveTab(selectedTabIndex);
+        }
+    }, []);
+
+    const handleClick = useCallback((index: number) => {
```
Comment (berezh) [inline src/components/tabs/index.tsx]: `useCallback` for all methods plz

Comment (akhlopiachyi) [reply src/components/tabs/index.tsx]: Done

File: src/components/tabs/index.scss [hunk 7]
```diff
@@ -0,0 +1,15 @@
+@import "src/components/style/base.scss";
+
+.tabs {
```
Comment (berezh) [inline src/components/tabs/index.scss:3]: Plz apply styles we to be look like on baseapp (wallet page). Not exact, but should be fealing that this is styled component

File: src/components/tabs/index.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    }, [children[activeTabIndex]]);
+
+    const renderActiveTabContent = useCallback(() => {
+        if (children[activeTabIndex]) {
+            return children[activeTabIndex].props.panel;
+        }
+    }, [children[activeTabIndex]]);
+
+    return (
+        <>
```
Comment (berezh) [inline src/components/tabs/index.tsx]: The root tag should contain main class of the component to have an impact on all children

File: src/components/tabs/index.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+    const renderChildrenWithTabsApiAsProps = useCallback(() => {
+        return Children.map(children, (child, index) => {
+            return cloneElement(child, {
+                tabIndex: index,
+                active: index === activeTabIndex,
+                onClick: handleClick,
+            });
+        });
+    }, [children[activeTabIndex]]);
```
Comment (berezh) [inline src/components/tabs/index.tsx:52]: Move it to `useMemo`

File: src/components/tabs/index.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    const handleClick = useCallback(
+        (index: number) => {
+            if (index !== activeTabIndex) {
+                setActiveTab(index);
+            }
+        },
+        [activeTabIndex]
+    );
+
+    const renderChildrenWithTabsApiAsProps = useCallback(() => {
```
Comment (berezh) [inline src/components/tabs/index.tsx]: Plz use more simple names, example `renderTabs`

File: src/components/tabs/index.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+                onClick: handleClick,
+            });
+        });
+    }, [children[activeTabIndex]]);
+
+    const renderActiveTabContent = useCallback(() => {
+        if (children[activeTabIndex]) {
+            return children[activeTabIndex].props.panel;
+        }
+    }, [children[activeTabIndex]]);
```
Comment (berezh) [inline src/components/tabs/index.tsx:58]: move to `useMemo`

File: src/components/tabs/index.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+}) => {
+    const [activeTabIndex, setActiveTab] = useState(0);
+
+    useEffect(() => {
+        if (
+            selectedTabIndex &&
+            selectedTabIndex > -1 &&
+            children[selectedTabIndex]
+        ) {
+            setActiveTab(selectedTabIndex);
```
Comment (berezh) [inline src/components/tabs/index.tsx]: why don't you move this initialialization to `useState` method?

**Review by berezh — CHANGES_REQUESTED** [review]

File: src/components/tabs/story/meta.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import { Meta } from '@storybook/react';
+import { Tabs } from 'src';
+
+export const createMeta = (title: string): Meta => {
+    return {
+        component: Tabs,
+        title,
+        argTypes: {
+            maxWidth: {
+                name: 'Content Width',
```
Comment (berezh) [inline src/components/tabs/story/meta.tsx]: No reason to have `Content Width` here.
Plz check how you can use Storybook controls for your component. You may add some stuff here and reduce stories you have.

File: src/components/tabs/index.tsx [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+            <div className={classNames('tabs__nav', tabNavClassName)}>
+                <div
+                    className={classNames(
+                        'tabs__nav-wrapper',
+                        tabNavWrapperClassName
+                    )}>
+                    {renderTabs}
+                </div>
+            </div>
+            <div className={classNames('tabs__panel', tabPanelClassName)}>
```
Comment (berezh) [inline src/components/tabs/index.tsx]: Plz don't go creazy with subclasses. Enough to have one root className to have impact to the rest children

Comment (akhlopiachyi) [reply src/components/tabs/index.tsx]: Done

File: src/components/tabs/index.scss [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+    &__nav {
+        background: var(--tabsHeaderBackgroundColor);
+        border-bottom: 1px solid var(--dividerColor);
+        margin: 0;
```
Comment (berezh) [inline src/components/tabs/index.scss]: do not add extra props. By default `div` always has `margin: 0; padding: 0`
Plz update all your SCSS files

Comment (akhlopiachyi) [reply src/components/tabs/index.scss]: Done

Comment (berezh) [reply src/components/tabs/index.scss]: I don't see changes

**Review by berezh — APPROVED** [review]

**Pull request merged**